lib/bson/parser/deserializer.js

```diff
@@ -173,6 +173,19 @@ var deserializeObject = function(buffer, index, options, isArray) {
         object[name] = deserializeObject(buffer, _index, options, false);
       }
 
+      if (name === 'dbref') {
```

Comment:
I think I'm confused.  This appears to be checking explicitly for a property called `dbref` on an object, that's not really spec-related is it?  This seems wrong, but maybe I'm misunderstanding the problem here.

Reply:
now that you say this and I look at the documentation on dbrefs I think this is wrong. 

I had just been looking at the test cases in [this file](https://github.com/mongodb/specifications/blob/master/source/bson-corpus/tests/dbref.json) and the fact that the last one is not supposed to be a dbref made me think dbrefs had to explicitly have that key. (since all the previous cases do).
but now that I compare it to the previous test case, I think it's actually because it has a $ key (`$banana`) which isn't part of a DBRef, but a non-$ additional key is ok.